src/components/layout/TopBar.jsx

```diff
@@ -18,6 +19,18 @@ const isEmailAddress = (input) => {
  * NavBar renders row of nav-items of form { name, icon, link }
  */
 class TopBar extends Component {
+  componentDidMount() {
+    // clear global store of expired banners
+    if (this.props.closedBanners) {
+      // eslint-disable-next-line no-restricted-syntax
+      for (const [key, value] of Object.entries(this.props.closedBanners)) {
+        if (Date.now() > value) {
+          this.props.onResetBanner({ id: key });
```

Comment:
I think this place is good for placing the logic but I didn't quite get the logic: we should only want to remove a banner from tthe store if the banner no longer exists in the config. If the banner still exists in the config and `Date.now() > its_reste_date`, the bannere should reappear right?

Reply:
the logic does not check the config, but rather checks the store for expired banners and removes them. If an admin removed a banner the users browser will still remember that the banner was closed until the reset date has been hit.

Reply:
Right, and that is what I'm confused at now. Doesn't this `resetMsgDays` determines the time period that we wait until this banner show up again? So under this assumption, if the current time is greater than `resetMsgDays` it doesn't means the banner is expired, it means it should re-appear. Or did I misunderstand this logic? 🤔

Reply:
you misunderstand, i think it will be easier if i just go threw the logic:

1. config is checked and displays banners that dont match record in local storage 
2. When user exits a banner object is stored in local storage that has timestamp of when it expires (current time plus resetMsgDays)
3. when user current time is greater than expiration time record is removed from local storage 
4. back to 1

Reply:
I see, thanks for the explaination, then I think we should update this descriptive text here https://github.com/uc-cdis/data-portal/pull/1398/files#diff-b44409708be4e154876122765a9fd3ada56184bb299149d53f7227a5db35c2eaR55 🤔 because it says the banner will re-appear

Reply:
banner will re appear once record is removed from local storage if its still in the config

Reply:
oh I see! thanks for the walk through 😅